src/Angor/Client/Pages/Wallet.razor

```diff
@@ -1255,4 +1257,48 @@ else
     {
         isShowExtraWord = !isShowExtraWord;
     }
-}
+
+    public async Task<IReadOnlyList<string>> GetBtcValuesInPreferredCurrency(params decimal[] btcBalances)
+    {
+        return await _currencyService.GetBtcValuesInPreferredCurrency(btcBalances);
+    }
+
+    public async Task<string> GetBtcValueInPreferredCurrency(decimal btcBalance)
+    {
+        var results = await GetBtcValuesInPreferredCurrency(btcBalance);
+        return results.FirstOrDefault() ?? "Error fetching value";
+    }
+
+
+    private async Task PrecomputeFiatValues()
+    {
+        fiatValues.Clear();
+
+        var addresses = accountBalanceInfo.AccountInfo.AllAddresses().ToList();
+        
+        if (addresses.Count == 0)
+        {
+            Logger.LogWarning("No addresses available for processing.");
+            return;
+        }
+
+        var btcBalances = addresses
```

Comment:
why not use the already existing type `accountBalanceInfo.TotalBalance` ?

Reply:
Use GetBtcValuesInPreferredCurrency for fiat conversion.